samples/CrawlFeaturizer/ActionFeaturizer/CognitiveTextAnalyticsFeaturizer.cs

```diff
@@ -27,21 +28,21 @@ internal CognitiveTextAnalyticsFeaturizer(CognitiveTextAnalyzer cognitiveTextAna
         public async Task FeaturizeActionsAsync(IEnumerable<CrawlAction> actions)
         {
             await Task.WhenAll(actions.Select(async a =>
-                {
-                    Metadata metadata = a.Metadata.ToObject<Metadata>();
-                    string content = $"{metadata.Title ?? string.Empty} {metadata.Description ?? string.Empty}";
-
-                    // Get key phrases from the article title and description
-                    IList<string> keyPhrases = await cognitiveTextAnalyzer.GetKeyPhrasesAsync(content);
-                    
-                    // Create a dictionary of key phrases (with a constant values) since at this time we do not support list of strings features.
-                    var keyPhrasesWithConstValues = keyPhrases.ToDictionary(x => x, x=>1);
-                    a.Features.Add(new { keyPhrases = keyPhrasesWithConstValues});
-
-                    // Get sentiment score for the article
-                    double? sentiment = await cognitiveTextAnalyzer.GetSentimentAsync(content);
-                    a.Features.Add(new { sentiment });
-                }
+            {
+                Metadata metadata = a.Metadata.ToObject<Metadata>();
+                string content = $"{metadata.Title ?? string.Empty} {metadata.Description ?? string.Empty}";
+
+                // Get key phrases from the article title and description
+                Response<KeyPhraseCollection> keyPhrases = await textAnalyticsClient.ExtractKeyPhrasesAsync(content);
+
+                // Create a dictionary of key phrases (with a constant values) since at this time we do not support list of strings features.
+                var keyPhrasesWithConstValues = keyPhrases.Value.ToDictionary(x => x, x => 1);
+                a.Features.Add(new { keyPhrases = keyPhrasesWithConstValues });
+
+                // Get sentiment score for the article
+                DocumentSentiment sentiment = await textAnalyticsClient.AnalyzeSentimentAsync(content);
+                a.Features.Add(new { sentiment.ConfidenceScores });
```

Comment:
I want to make sure that the Object ConfidenceScores contains the information you actually need for this sample, and I don't see them being used in the code.
Just FYI, between versions 2 and 3 (current), Text Analytics changed the concept of the sentiment score. In v3, the DocumentSentiment.Sentiment contains the sentiment of the document, and the DocumentSentiment.ConfidenceScores contains the softmax value for each possible Sentiment.
Because of this relationship, the Confidence's scores are better use when next to the Sentiment label than by themselves.

More information => https://docs.microsoft.com/en-us/azure/cognitive-services/Text-Analytics/how-tos/text-analytics-how-to-sentiment-analysis?tabs=version-3-1#sentiment-analysis

Reply:
The information of ConfidenceScore actually used at line 116: Console.WriteLine("Features : " + JsonSerializer.Serialize(**recommendedAction.Features**, options)); ConfidenceScore part contains sentiment. It is much better to print out here.
![image](https://user-images.githubusercontent.com/65536078/100053660-75e86c00-2e5b-11eb-852a-6685b425ccc2.png)